google/cloud/sql/connector/connector.py

```diff
@@ -330,10 +330,17 @@ async def connect_async(
 
             # async drivers are unblocking and can be awaited directly
             if driver in ASYNC_DRIVERS:
-                return await connector(ip_address, instance_data.context, **kwargs)
+                return await connector(
+                    ip_address,
+                    instance_data.create_ssl_context(enable_iam_auth),
```

Comment:
i guess the downside of this is now we are creating an SSL/TLS context per connection vs in the past where we were creating it once per `ConnectionInfo` class...

Reply:
Is there a significant cost associated with creating an SSL context per connection? What would a Python developer expect?